### PR TITLE
Less buffer macros

### PR DESCRIPTION
**Pull request by ShelbyZ**
- Adding ref(buf, index) to socket.h
- removing RBUF* macros for ref<T>
- removing WBUF* macros for ref<T>
- fixing changes resulting from

### CODE REVIEW

File: src/map/lua/luautils.cpp [hunk 1]
```diff
@@ -1771,27 +1771,40 @@ namespace luautils
 
     int32 OnEffectTick(CBattleEntity* PEntity, CStatusEffect* PStatusEffect)
     {
-        lua_prepscript("scripts/%s.lua", PStatusEffect->GetName());
-
-        if (prepFile(File, "onEffectTick"))
+        try
```
Comment (teschnei) [inline src/map/lua/luautils.cpp]: unintentional testing code?

Comment (ShelbyZ) [reply src/map/lua/luautils.cpp]: yeah, leftovers frame last night

File: src/common/socket.h [hunk 2]
```diff
@@ -303,5 +290,16 @@ extern int32 naddr_;   // # of ip addresses
 	int32 sendudp(int32 fd,void *buff,size_t nbytes,int32 flags,const struct sockaddr *from,socklen_t addrlen);
 #endif 
 
+template<typename T, typename U>
+T& ref(U* buf, std::size_t index)
+{
+    return *reinterpret_cast<T*>(buf + index);
+}
+
+template<typename T, typename U = void*>
```
Comment (teschnei) [inline src/common/socket.h]: what is this?

Comment (ShelbyZ) [reply src/common/socket.h]:
```cpp
void* thing;
thing + 1; // error, void* unknown size
```

If I don't provide the overload, it means adding more (uint8*) into message, login_auth, etc.  Where we are using zmq and get data() (void*).

**Comment by teschnei** [comment]
but why have a template parameter U if you don't use it anywhere?

**Comment by teschnei** [comment]
Also... we should cast there because it's a void pointer, in my opinion

**Comment by ShelbyZ** [comment]
Is the suggestion to move to:

```cpp
template<typename T>
T& ref(void* buf, std::size_t index)
{
    return *reinterpret_cast<T*>((uint8*)buf + index);
}
```

**Comment by teschnei** [comment]
I'd rather just get rid of it entirely and cast void pointers to uint8
pointers.  We could theoretically want a buffer that's not uint8...

There's not that many void pointers anyways I don't think.  Should be just
the zmq buffers which we can't control the type of

**Pull request merged**